luigi/scheduler.py

```diff
@@ -318,6 +323,18 @@ def __init__(self, task_id, status, deps, resources=None, priority=0, family='',
     def __repr__(self):
         return "Task(%r)" % vars(self)
 
+    @property
+    def params(self):
+        return self.__params
+
+    @params.setter
+    def params(self, params):
```

Comment:
I haven't seen this `@var.setter` syntax a lot in luigi. Maybe it's better to just call it `set_params` to make it clear it's more than an assignment? Assignments that aren't O(1) feels very unintuitive.